### ISSUE TITLE
StackOverflowError on BasicRobot.findOuterPopupMenu()

### ISSUE THREAD

**Issue by aguelle** [joel-costigliola/assertj-swing#180]
Thanx for the great work on 3.1.1. Here is a new issue I found.

```
public class TableFrame extends JFrame {

	public TableFrame() {

		setSize(200, 100);
		setTitle(getClass().getCanonicalName());
		setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);

		JTable table = new JTable();
		add(table);

		table.addMouseListener(new MouseAdapter() {

			@Override
			public void mousePressed(MouseEvent e) {
				mouseReleased(e);
			}

			@Override
			public void mouseReleased(MouseEvent e) {
				new PopupMenu();
			}

		});

		setVisible(true);
	}

	public static void main(String[] args) {
		new TableFrame();

	}

	private class PopupMenu extends JPopupMenu {

		public PopupMenu() {

			JMenuItem menuItem = new JMenuItem("TEST");
			add(menuItem);
			setVisible(true);
		}
	}
}

import org.assertj.swing.edt.GuiActionRunner;
import org.assertj.swing.edt.GuiQuery;
import org.assertj.swing.fixture.FrameFixture;
import org.assertj.swing.junit.testcase.AssertJSwingJUnitTestCase;
import org.junit.Test;

public class TableTest extends AssertJSwingJUnitTestCase {

	protected FrameFixture window;

	@Override
	protected void onSetUp() {

		TableFrame mainFrame = GuiActionRunner
				.execute(new GuiQuery<TableFrame>() {

					protected TableFrame executeInEDT() {

						return new TableFrame();
					}
				});

		window = new FrameFixture(robot(), mainFrame);
	}

	@Test
	public void openPopup() {
		window.table().showPopupMenu();
	}

}
```

Running the Test "openPopup()" results in:

```
java.lang.StackOverflowError
	at java.util.HashMap$KeyIterator.<init>(HashMap.java:1451)
	at java.util.HashMap$KeySet.iterator(HashMap.java:912)
	at java.util.HashSet.iterator(HashSet.java:172)
	at java.util.AbstractCollection.toArray(AbstractCollection.java:137)
	at java.util.ArrayList.addAll(ArrayList.java:577)
	at org.assertj.swing.core.BasicRobot.findOuterPopupMenu(BasicRobot.java:913)
	at org.assertj.swing.core.BasicRobot.findOuterPopupMenu(BasicRobot.java:916)
```

**Comment by croesch** [comment]
Okay thanks.

But in your code there seems to be little problem which fixes the StackOverflowError. See [DisabledTableFrame.java](https://github.com/joel-costigliola/assertj-examples/blob/master/assertj-swing-aut/src/main/java/org/assertj/swing/aut/components/DisabledTableFrame.java). You need to add: ``e.isPopupTrigger()``

```
Note: Popup menus are triggered differently on different systems. Therefore, isPopupTrigger should be checked in both mousePressed and mouseReleased for proper cross-platform functionality.
```

Anyway, I see the problem in AssertJ Swing's code and I'm going to improve this.

**Comment by croesch** [comment]
Included in 3.2.0.

**Comment by aguelle** [comment]
Thanx Christian for the remark on popup menus. Great work.
